homeassistant/components/concord232/alarm_control_panel.py

```diff
@@ -124,14 +134,41 @@ def alarm_arm_home(self, code: str | None = None) -> None:
             self._alarm.arm("stay", "silent")
         else:
             self._alarm.arm("stay")
+        self.wait_for_state(STATE_ALARM_ARMED_HOME)
 
     def alarm_arm_away(self, code: str | None = None) -> None:
         """Send arm away command."""
         if not self._validate_code(code, STATE_ALARM_ARMED_AWAY):
             return
         self._alarm.arm("away")
+        self.wait_for_state(STATE_ALARM_ARMED_AWAY)
+
+    def wait_for_state(self, target_state: str, timeout: int = 5) -> None:
+        """Wait for alarm panel to arm/disarm."""
+        if target_state.startswith("disarm"):
```

Comment:
Why not check check if target_state == STATE_ALARM_DISARMED ?

Reply:
Makes sense. Change made.